cpp/src/io/parquet/stats_filter_helpers.cpp

```diff
@@ -48,12 +49,66 @@ std::reference_wrapper<ast::expression const> stats_expression_converter::visit(
   return expr;
 }
 
-std::reference_wrapper<ast::expression const> stats_expression_converter::visit(
+std::reference_wrapper<ast::expression const> stats_columns_collector::visit(
   ast::column_name_reference const& expr)
 {
   CUDF_FAIL("Column name reference is not supported in statistics AST");
 }
 
+std::reference_wrapper<ast::expression const> stats_columns_collector::visit(
+  ast::operation const& expr)
+{
+  using cudf::ast::ast_operator;
+  auto const operands = expr.get_operands();
+  auto const op       = expr.get_operator();
+
+  if (auto* v = dynamic_cast<ast::column_reference const*>(&operands[0].get())) {
```

Comment:
Identical to `stats_expression_converter::visit()` function except we just set the `mask[column_index]` and move on.